File: backend/internal/models/snippetModel.go
package models

type Snippet struct {
	SnippetID   uint     `json:"snippet_id" gorm:"primaryKey;autoincrement"`
	Title       string   `json:"title"`
	Summary     string   `json:"summary" gorm:"unique"`
	Description string   `json:"description"`
	Example     string   `json:"example"`
	Tags        string   `json:"tags"`
	LanguageID  uint     `json:"language_fk"`
	Language    Language `json:"-"`
}

func CreateSnippet(
	snippetJson map[string]interface{}, language string) *Snippet {

	languageList := GetLanguageList()

	return &Snippet{
		Title:       snippetJson["title"].(string),
		Summary:     snippetJson["summary"].(string),
		Description: snippetJson["description"].(string),
		Example:     snippetJson["example"].(string),
		Tags:        snippetJson["tags"].(string),
		LanguageID:  languageList[language].LanguageID,
		Language:    *languageList[language],
	}
}

func (s *Snippet) GetList() []string {
	return []string{
		s.Title,
		s.Summary,
		s.Description,
		s.Example,
		s.Tags,
	}
}

File: frontend/snipbin/src/models/snippetModel.ts
import { fetchApi, fetchFromPostApi } from "../util/apiHandler"

export interface SnippetModel {
    snippet_id: number
    title: string
    summary: string
    description: string
    example: string
    tags: string
    language_fk: number
}

export interface LanguageModel {
    language_id: number
    language_name: string
}

export class SnippetApis {
    // snippet!: Array<SnippetModel>;
    url = 'http://localhost:8080/api/v1'

    fetchAllSnippets(language: string): Promise<SnippetModel[]> {
        const fetchSnippetUrl = `${this.url}/${language}`
        return fetchApi<SnippetModel[]>(fetchSnippetUrl)
    }

    fetchSearchResult(language: string, searchString: string): Promise<SnippetModel[]> {
        const bodyJson: {searchString : string} = {"searchString" : searchString}
        const SearchUrl = `${this.url}/${language}/search`
        return fetchFromPostApi<SnippetModel[]>(SearchUrl, bodyJson)
    }

    fetchLanguages(): Promise<LanguageModel[]> {
        const fetchLanguageUrl = `${this.url}/languagelist`
        return fetchApi<LanguageModel[]>(fetchLanguageUrl)
    }

    fetchSnippet(language: string, snippetTitle: string): Promise<SnippetModel> {
        const fetchSnippetUrl = `${this.url}/${language}/${snippetTitle}`
        return fetchApi<SnippetModel>(fetchSnippetUrl) 
    }
}